config/config.go

```diff
@@ -0,0 +1,181 @@
+package config
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+)
+
+type Application struct {
+	FilePath,
+	Postcode,
+	StartTimeString,
+	EndTimeString string
+	KeywordsString []string
+	StartTime,
+	EndTime int
+}
+
+func Init() *Application {
+	var config Application
+
+	var keywordsString, postcode, timeWindow, path string
+	flag.StringVar(&keywordsString, "keywords", "Potato,Veggie,Mushroom", "Comma-separated list of keywords to search for in recipe names")
+	flag.StringVar(&postcode, "postcode", "10120", "Postcode to search for")
+	flag.StringVar(&timeWindow, "time", "10AM-3PM", "Time window to search for in format 'start-end' (e.g., '10AM-3PM')")
+	flag.StringVar(&path, "file", "deliveries.json", "Path to file location")
+	flag.Parse()
+
+	err := validatePostcode(postcode)
+	if err != nil {
+		log.Println("Error:", err)
+		return nil
+	}
+	keywords, err := validateKeywords(keywordsString)
+	if err != nil {
+		log.Println("Error:", err)
+		return nil
+	}
+	err = validateTimeWindow(timeWindow)
+	if err != nil {
+		log.Println("Error:", err)
+		return nil
+	}
+	err = validateFilePath(path)
+	if err != nil {
+		log.Println("Error:", err)
+		return nil
+	}
+	config.KeywordsString = keywords
+	config.Postcode = postcode
+	timeParts := strings.Split(timeWindow, "-")
+	if len(timeParts) != 2 {
+		log.Println("Invalid time window format. Expected 'start-end', got", timeWindow)
+		return nil
+	}
+	config.StartTimeString = timeParts[0]
+	config.EndTimeString = timeParts[1]
+
+	config.StartTime = ParseHour(config.StartTimeString)
+	config.EndTime = ParseHour(config.EndTimeString)
+	config.FilePath = path
+	return &config
+
+}
+
+func validateFilePath(path string) error {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("file does not exist: %s", path)
+	} else if err != nil {
+		return fmt.Errorf("error accessing file: %s", err)
+	}
+	return nil
+}
+
+func ParseHour(timeString string) int {
+	hour, _ := strconv.Atoi(timeString[:len(timeString)-2])
+	if strings.HasSuffix(timeString, "PM") && hour != 12 {
+		hour += 12
+	}
+	if strings.HasSuffix(timeString, "AM") && hour == 12 {
+		hour = 0
+	}
+	return hour
+}
+func validateKeywords(keywords string) ([]string, error) {
+	// Split the keywords string into a slice
```

Comment:
как считаешь не будет ли оптимальнее сделать так
```suggestion
	keywords = strings.TrimSpace(strings.ToLower(keywords))
	// Split the keywords string into a slice
```

Reply:
хотя тут  тогда сломаются пробелы в ингридиентах